packages/opentelemetry-core/src/distribute-context/propagation/HttpTraceContext.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Carrier,
+  Context,
+  HttpTextFormat,
+  DistributedContext,
+} from '@opentelemetry/api';
+
+import {
+  getDistributedContext,
+  setDistributedContext,
+} from '../distribute-context';
+
+export const CORRELATION_CONTEXT_HEADER = 'Correlation-Context';
+
+/**
+ * Propagates {@link DistributedContext} through Context format propagation.
+ *
+ * Based on the Correlation Context specification:
+ * https://w3c.github.io/correlation-context/
+ */
+export class HttpTraceContext implements HttpTextFormat {
+  inject(context: Context, carrier: Carrier) {
+    const distContext = getDistributedContext(context);
+    if (distContext) {
+      const headerValue = Object.keys(distContext)
+        .map(
+          (key: string) =>
+            `${encodeURIComponent(key)}=${encodeURIComponent(
+              distContext[key].value
+            )}`
+        )
+        .join(',');
+      if (headerValue.length > 0) {
+        carrier[CORRELATION_CONTEXT_HEADER] = headerValue;
+      }
+    }
+  }
+
+  extract(context: Context, carrier: Carrier): Context {
+    const headerValue: string = carrier[CORRELATION_CONTEXT_HEADER] as string;
```

Comment:
http headers are case intensive in general so this won't work in all cases.
It will for sure be problematic with node.js built in http and http2 modules as they give provide incoming headers always in lowercase (see https://nodejs.org/dist/latest-v12.x/docs/api/http.html#http_http).